external/kafka-0-10/src/main/scala/org/apache/spark/streaming/kafka010/KafkaUtils.scala

```diff
@@ -184,10 +184,10 @@ object KafkaUtils extends Logging {
    */
   private[kafka010] def fixKafkaParams(kafkaParams: ju.HashMap[String, Object]): Unit = {
     logWarning(s"overriding ${ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG} to false for executor")
-    kafkaParams.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false: java.lang.Boolean)
+    kafkaParams.putIfAbsent(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false: java.lang.Boolean)
 
     logWarning(s"overriding ${ConsumerConfig.AUTO_OFFSET_RESET_CONFIG} to none for executor")
-    kafkaParams.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "none")
+    kafkaParams.putIfAbsent(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "none")
```

Comment:
I think the Spark doc https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html#kafka-specific-configurations explains this.